flatland_plugins/test/dynamics_limits_test.cpp

```diff
@@ -0,0 +1,316 @@
+  dynamics_limits.Configure(config);
+
+  // Ensure params are correct
+  EXPECT_NEAR(0.1, dynamics_limits.acceleration_limit_, 1e-3);
+  EXPECT_NEAR(0.2, dynamics_limits.deceleration_limit_, 1e-3);
+  EXPECT_NEAR(2.0, dynamics_limits.velocity_limit_, 1e-3);
+}
+
+// todo: test Saturate
+
+/**
+ * Test Limit function without dynamics limits in place
+ */
+TEST(DynamicsLimitsTest, test_Limit_noop) {
+  auto dynamics_limits = DynamicsLimits();
+
+  double result;
+  result = dynamics_limits.Limit(0, 1.0, 0.05);
+  EXPECT_NEAR(1.0, result, 1e-3);
+  result = dynamics_limits.Limit(1.0, 1.0, 0.05);
+  EXPECT_NEAR(1.0, result, 1e-3);
+  result = dynamics_limits.Limit(0.1, 1.0, 0.05);
+  EXPECT_NEAR(1.0, result, 1e-3);
+  result = dynamics_limits.Limit(-100, 1.0, 0.1);
+  EXPECT_NEAR(1.0, result, 1e-3);
+  result = dynamics_limits.Limit(-100, -1.0, 0.1);
+  EXPECT_NEAR(-1.0, result, 1e-3);
+  result = dynamics_limits.Limit(-100, -21.0, 0.1);
+  EXPECT_NEAR(-21.0, result, 1e-3);
+}
+
+
+/**
+ * Test Limit function
+ */
+TEST(DynamicsLimitsTest, test_Limit_velocity) {
+  auto dynamics_limits = DynamicsLimits();
+
+  YAML::Node config = YAML::Node();
+  config["velocity_limit"] = 1.0;
+  dynamics_limits.Configure(config);
+
+  EXPECT_NEAR(0.0, dynamics_limits.acceleration_limit_, 1e-3);
+  EXPECT_NEAR(0.0, dynamics_limits.deceleration_limit_, 1e-3);
+  EXPECT_NEAR(1.0, dynamics_limits.velocity_limit_, 1e-3);
+
+  double result;
+  result = dynamics_limits.Limit(0, 1.0, 0.05);
+  EXPECT_NEAR(1.0, result, 1e-3);
+  result = dynamics_limits.Limit(1.0, 1.0, 0.05);
+  EXPECT_NEAR(1.0, result, 1e-3);
+  result = dynamics_limits.Limit(0, 2.0, 0.05);
+  EXPECT_NEAR(1.0, result, 1e-3);
+  result = dynamics_limits.Limit(0, -1.0, 0.1);
+  EXPECT_NEAR(-1.0, result, 1e-3);
+  result = dynamics_limits.Limit(0, -0.5, 0.1);
+  EXPECT_NEAR(-0.5, result, 1e-3);
+  result = dynamics_limits.Limit(1.0, -0.5, 0.1);
+  EXPECT_NEAR(-0.5, result, 1e-3);
+  result = dynamics_limits.Limit(-100, -1.0, 0.1);
+  EXPECT_NEAR(-1.0, result, 1e-3);
+  result = dynamics_limits.Limit(0, -3.0, 0.1);
+  EXPECT_NEAR(-1.0, result, 1e-3);
+}
+
+
+/**
+ * Test Limit function wrt. acceleration
+ */
+TEST(DynamicsLimitsTest, test_Limit_acceleration) {
+  auto dynamics_limits = DynamicsLimits();
+
+  YAML::Node config = YAML::Node();
+  config["acceleration_limit"] = 9.0;
+  dynamics_limits.Configure(config);
+
+  EXPECT_NEAR(9.0, dynamics_limits.acceleration_limit_, 1e-3);
+  EXPECT_NEAR(9.0, dynamics_limits.deceleration_limit_, 1e-3);
+  EXPECT_NEAR(0.0, dynamics_limits.velocity_limit_, 1e-3);
+
+  double result;
+  result = dynamics_limits.Limit(0, 1.0, 0.05);  // Try to accelerate at 20m/s/s, when the limit is 9
+  EXPECT_NEAR(0.45, result, 1e-3);
+  result = dynamics_limits.Limit(1.0, 0.0, 0.05);  // Try to deccelerate at -20m/s/s, when the limit is 9
+  EXPECT_NEAR(0.55, result, 1e-3);
+  result = dynamics_limits.Limit(10.0, 0.0, 0.05);  // Try to deccelerate at -20m/s/s, when the limit is 9
```

Comment:
Comment should say -200 m/s/s?

Reply:
I'd thought this was 20m/s/s? We're trying to go from 1m/s to 0m/s in 0.05 seconds (0m/s-1m/s)/0.05s = 20m/s/s?

Reply:
Some of the examples have a 10 m/s range, like this and the next 2, wasn't sure if that meant the deceleration was 10x of the previous example which is a 1 m/s range.

Reply:
oh, sorry, I was hung up on line 193, and missed 195. You're right!